tests/Microsoft.Bot.Connector.Tests/OAuthConnectorTests.cs

```diff
@@ -59,45 +86,241 @@ public async Task GetUserToken_ShouldReturnNullOnInvalidConnectionString()
+        public async Task GetSignInUrlWithHttpMessagesAsync_ShouldThrowOnNoLocalBot()
+        {
+            var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("token"));
+            ServiceClientTracing.IsEnabled = true;
+
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("AGENT_OS")) &&
+                Environment.GetEnvironmentVariable("AGENT_OS").Equals("Windows_NT"))
+            {
+                // Automated Windows build exception:
+                await Assert.ThrowsAsync<Microsoft.Rest.HttpOperationException>(() => client.BotSignIn.GetSignInUrlAsync(
+                    "dummyState", "dummyCodeChallenge", "dummyEmulatorUrl", "dummyFinalRedirect"));
+            }
+            else
+            {
+                // MacLinux build and local build exception:
+                await Assert.ThrowsAsync<System.Net.Http.HttpRequestException>(() => client.BotSignIn.GetSignInUrlAsync(
+                    "dummyState", "dummyCodeChallenge", "dummyEmulatorUrl", "dummyFinalRedirect"));
+            }
+        }
+
+        [Fact]
+        public async Task GetTokenStatusWithHttpMessagesAsync_ShouldThrowOnNullUserId()
         {
             var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("token"));
             await Assert.ThrowsAsync<ValidationException>(() => client.UserToken.GetTokenStatusAsync(null));
         }
 
         [Fact]
-        public async Task GetAadTokensAsync_ShouldThrowOnNullUserId()
+        public async Task GetTokenStatusWithHttpMessagesAsync_ShouldThrowOnNoLocalBot()
+        {
+            var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("token"));
+            ServiceClientTracing.IsEnabled = true;
+ 
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("AGENT_OS")) &&
+                Environment.GetEnvironmentVariable("AGENT_OS").Equals("Windows_NT"))
+            {
+                // Automated Windows build exception:
+                await Assert.ThrowsAsync<Microsoft.Bot.Schema.ErrorResponseException>(() => client.UserToken.GetTokenStatusAsync(
+                    "dummyUserId", "dummyChannelId", "dummyInclude"));
+            }
+            else
+            {
+                // MacLinux build and local build exception:
+                await Assert.ThrowsAsync<System.Net.Http.HttpRequestException>(() => client.UserToken.GetTokenStatusAsync(
+                    "dummyUserId", "dummyChannelId", "dummyInclude"));
+            }
+        }
+
+        [Fact]
+        public async Task GetAadTokensWithHttpMessagesAsync_ShouldThrowOnNullUserId()
         {
             var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("token"));
             await Assert.ThrowsAsync<ValidationException>(() => client.UserToken.GetAadTokensAsync(null, "connection", new AadResourceUrls() { ResourceUrls = new string[] { "hello" } }));
```

Comment:
Should we refactor into a helper method to avoid code repetition?

Reply:
To replace one line of code with a call to a helper seems equally repetitive to me, and more complex.